test/gurka/test_shortcut.cc

```diff
@@ -289,3 +289,81 @@ TEST(Shortcuts, ShortcutsInBins) {
 
   EXPECT_EQ(shortcut_cnt, 1);
 }
+
+TEST(Shortcuts, ShortcutRestrictions) {
+  const std::string ascii_map = R"(
+  A--B--C--D--E--F
+
+  G--H--I--J--K--L
+  )";
+
+  // the first line should produce only one shortcut, the second two
+  gurka::ways ways = {
+      {"AB", {{"highway", "motorway"}, {"name", "highway"}}},
+      {"BC", {{"highway", "motorway"}, {"name", "highway"}}},
+      {"CD", {{"highway", "motorway"}, {"name", "highway"}, {"maxweight", "3.5"}}},
+      {"DE", {{"highway", "motorway"}, {"name", "highway"}, {"maxweight", "8"}}},
+      {"EF", {{"highway", "motorway"}, {"name", "highway"}}},
+
+      {"GH", {{"highway", "motorway"}, {"name", "highway"}}},
+      {"HI", {{"highway", "motorway"}, {"name", "highway"}, {"hazmat", "yes"}}},
+      {"IJ", {{"highway", "motorway"}, {"name", "highway"}, {"hazmat", "yes"}}},
```

Comment:
also that we're building shortcuts over consecutive edges with the same restriction. could've done many more permutations, but this should be ok